src/System.Private.CoreLib/shared/System/Runtime/InteropServices/SafeHandle.cs

```diff
@@ -0,0 +1,249 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.ConstrainedExecution;
+using System.Threading;
+
+namespace System.Runtime.InteropServices
+{
+    // This implementation does not employ critical execution regions and thus cannot
+    // reliably guarantee handle release in the face of thread aborts.
+
+    /// <summary>Represents a wrapper class for operating system handles.</summary>
+    public abstract partial class SafeHandle : CriticalFinalizerObject, IDisposable
+    {
+        // IMPORTANT:
+        // - Do not add or rearrange fields as the EE depends on this layout,
+        //   as well as on the values of the StateBits flags.
+        // - The EE may also perform the same operations using equivalent native
+        //   code, so this managed code must not assume it is the only code
+        //   manipulating _state.
+
+        /// <summary>Specifies the handle to be wrapped.</summary>
+        [SuppressMessage("Microsoft.Security", "CA2111:PointersShouldNotBeVisible")]
+        protected IntPtr handle;
+        /// <summary>Combined ref count and closed/disposed flags (so we can atomically modify them).</summary>
+        private volatile int _state = StateBits.RefCountOne; // Ref count 1 and not closed or disposed.
```

Comment:
Would be more readable to have the initialization in ctor

Reply:
I find this more readable, with the ctors reserved only for things that must be there, but I can change it.